src/data_race.rs

```diff
@@ -542,6 +542,35 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriEvalContextExt<'mir, 'tcx> {
         Ok(old)
     }
 
+    /// Perform an conditional atomic exchange with a memory place and a new
+    /// scalar value, the old value is returned.
+    fn atomic_min_max_scalar(
+        &mut self,
+        place: MPlaceTy<'tcx, Tag>,
```

Comment:
out of performance reasons, lots of code moved to using `&MplaceTy` and `&OpTy`, please also use this here

Reply:
@oli-obk So `place` would be a `&OpTy<'tcx, Tag>`

Reply:
`&MPlaceTy` makes more sense I think.